feat: add single GTFS Feed and GTFS-RT Feed pages

indraneel opened the pull request:
**Summary:**

Creates `Feed.tsx` to match individual GTFS Schedule feed page to [design ](https://www.figma.com/file/LZVQecGplb4Lnuh7a24Jef/GTFS-Tool-Forms?type=design&node-id=1063-857&mode=dev). and match [Match GTFS Realtime individual feed page](https://www.figma.com/file/LZVQecGplb4Lnuh7a24Jef/GTFS-Tool-Forms?type=design&node-id=1254%3A1743&mode=design&t=ztQe5LNqVxOuqd8y-1). Closes #352 and #353

**Expected behavior:** 

![image](https://github.com/MobilityData/mobility-feed-api/assets/444765/d9557504-38ac-47a7-8c4b-3734d703ed3c)

![image](https://github.com/MobilityData/mobility-feed-api/assets/444765/22cab6f9-3839-48f3-b090-5d695483f543)

**Testing tips:**

Navigate manually to `/feeds/<mdb-##>`

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [x] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

Alessandro100 commented on web-app/src/app/screens/Feed.tsx:
```diff
@@ -0,0 +1,603 @@
+                        {latestDataset?.validation_report?.features?.map(
+                          (v) => (
+                            <TableRow key={v}>
+                              <TableCell>{v}</TableCell>
```
format feature name using `DATASET_FEATURES_FILES_MAPPING`  + add second column of "File or Field Associated"

indraneel replied:
Based on my conversation with @emmambd, I'm under the impression we are omitting "File or Field Associated"

emmambd replied:
Correct, but the feature names themselves will be used

indraneel replied:
For `mdb-123` does this look correct? 
![image](https://github.com/MobilityData/mobility-feed-api/assets/444765/650c6bd8-64de-4c82-aae7-23b5842bd872)

Alessandro100 commented on web-app/src/app/screens/Feed.tsx:
This file is quite large, it would be great to see it broken down into smaller components

indraneel replied:
Sure - do you have a sense of how you might prefer this to be split? Logically I can imagine splitting the render function into:
- `renderGtfsFeedPage`
- `renderGtfsRTFeedPage`

Alessandro100 replied:
I see that GtfsFeed and RT share a fair amount of code, maybe we can have components for elements on the page `FeedSummary`, `Datasets` etc

indraneel replied:
Pushed changes with functions that render the following:

- `renderFeedSummary`
- `renderDataQualitySummary`
- `renderFeaturesList`
- `renderAssociatedGTFSFeed`
- `renderPreviousDatasets`

Alessandro100 commented on web-app/src/app/screens/Feed.tsx:
We should handle the error state if a call fails, do you want to add it in this PR or create a new ticket / task?

indraneel replied:
I can handle a basic error state here. If we need more complex design / changes I think that would be better suited for a different PR.

Alessandro100 replied:
a basic error state is perfect for now. @emmambd do you see a world where we have specific error states ex: "Error making the http call" and "Error: This feed Id does not exist" or just general "Error, go back home"

emmambd replied:
@Alessandro100 I think basic error state is sufficient for now. I can't think of a common use case where we'd need something more complex in a situation where the feed itself won't display.

indraneel replied:
added a basic error state w/ the following copy
![image](https://github.com/MobilityData/mobility-feed-api/assets/444765/eb3c9458-dee5-4e86-b08f-6149abc40180)

Alessandro100 commented on web-app/src/app/screens/Feed.tsx:
```diff
@@ -0,0 +1,561 @@
+                  endIcon={<LaunchOutlined />}
+                >
+                  <a
+                    href={feed?.source_info?.license_url}
```
If the `license_url` is empty, I would just hide this button

Alessandro100 commented on web-app/src/app/screens/Feed.tsx:
```diff
@@ -0,0 +1,561 @@
+                  endIcon={<LaunchOutlined />}
+                >
+                  <a
+                    href={feed?.source_info?.authentication_info_url}
```
If `authentication_info_url` is empty I would hide this button as well

Alessandro100 commented on web-app/src/app/screens/Feed.tsx:
```diff
@@ -0,0 +1,561 @@
+    }
+  }, []);
+
+  const renderFeedSummary = (): JSX.Element => (
```
I love the methods made to render this content, it makes the code much more structured. I'd like to see these methods in their own component to bring this file from ~600 lines to ~250 lines

indraneel replied:
Pushed this change.

indraneel commented:
> @indraneel It looks like there's some issues with the realtime vs. schedule querying - I can see the page load a realtime page before going to the GTFS Schedule feed:
> 
>  Screen.Recording.2024-05-22.at.8.44.38.AM.mov 
> It also looks like when a URL doesn't exist it's defaulting to some kind of empty link - can we remove the fields when the URL isn't present?
> 
>  Screen.Recording.2024-05-22.at.8.47.43.AM.mov

I pushed changes addressing these issues. Thanks  for spotting them @emmambd

Alessandro100 reviewed (approved):
Awesome changes, this looks great! Just merge the latest from main and it's good to go!

emmambd commented:
Looks like the feature flag wasn't set here, so the empty feeds page is on prod. #477. Should use and check for feature flags being enabled for front end work after this.
